src/lib/mina_lib/mina_lib.ml

```diff
@@ -2829,24 +2844,83 @@ module Hardfork_config = struct
   type inputs =
     { source_ledgers : genesis_source_ledgers
     ; global_slot_since_genesis : Mina_numbers.Global_slot_since_genesis.t
+    ; genesis_state_timestamp : string
     ; state_hash : State_hash.t
     ; staking_epoch_seed : Epoch_seed.t
     ; next_epoch_seed : Epoch_seed.t
     ; blockchain_length : Mina_numbers.Length.t
-    ; block_timestamp : Block_time.t
     }
 
+  (** The genesis state timestamp string is the timestamp of the start of the
```

Comment:
Suggest:
Replace 
```
the timestamp of the start of the [global_slot] of the hard fork
```
with 
```
the timestamp of the start of the epoch of the [global_slot]
```

 I don't think anything implementation-wise here is related to HF, although it's probably used in that context. 

Also I'd suggest rename the input from `global_slot` -> `global_slot_since_hard_fork` just so I don't need to look at type annotations.